gm4_mountaineering/data/gm4_mountaineering/functions/ski_effects/skis_equipped.mcfunction

```diff
@@ -0,0 +1,28 @@
+# run all ski-related functions
+# @s = player wearing skis
+# at @s
+# run from main
+
+# Adds tags for whether the player is skiing normally or skiing on a non-snowy block
+execute run function gm4_mountaineering:ski_effects/check_snowy_block
+
+# Records the player's y-velocity in a scoreboard
+execute run function gm4_mountaineering:get_velocity
+
+# Adds speed effects for skiing on a flat surface
+execute if entity @s[tag=gm4_mountaineering_skiing] run function gm4_mountaineering:ski_effects/flat_ski_speed/add_flat_ski_speed
+execute if entity @s[tag=!gm4_mountaineering_skiing] run function gm4_mountaineering:ski_effects/flat_ski_speed/remove_flat_ski_speed
```

Comment:
You do this quite a few more times, but I'll leave the comment here. These functions will keep running every clock cycle. You should add a tag to the player in `add_flat_ski_speed`, I'll use `gm4_mountaineering_ski_speed` as an example, and remove it in `remove_flat_ski_speed`
```suggestion
execute if entity @s[tag=gm4_mountaineering_skiing,tag=!gm4_mountaineering_ski_speed] run function gm4_mountaineering:ski_effects/flat_ski_speed/add_flat_ski_speed
execute if entity @s[tag=!gm4_mountaineering_skiing,tag=gm4_mountaineering_ski_speed] run function gm4_mountaineering:ski_effects/flat_ski_speed/remove_flat_ski_speed
```